validator/efotrait.py

```diff
@@ -1,17 +1,17 @@
-import requests
+from validator.request.connector import Connector, ConnectorException
+
 
 class EFOTrait():
 
     def __init__(self, id):
         self.id = id
         self.label = None
 
-    def populate_from_efo(self):
-        response = requests.get('https://www.ebi.ac.uk/ols/api/ontologies/efo/terms?obo_id=%s'%self.id.replace('_', ':'))
-        if response.status_code == 200:
-            response = response.json()['_embedded']['terms']
-            if len(response) == 1:
-                response = response[0]
-                self.label = response['label']
-                return True
-        return False
\ No newline at end of file
+    def populate_from_efo(self, connector: Connector):
```

Comment:
Same here with `DefaultConnector` there instead of the abstract `Connector` ?

Reply:
No problem, it makes sense to use abstract classes. I was just wondering if, nothing is passed as `connector` parameter, the DefaultConnector should be used instead of Connector.
I guess that the `connector` parameter is mandatory anyway so Connector should be fine then.

Reply:
The class PGSMetadataValidator has DefaultConnector as default value:
https://github.com/fyvon/pgs_template_validator/blob/63cb5f297429c4c6ac5bf7757a78c8ee68a9c6d3/validator/main_validator.py#L63-L65
And this connector is the one subsequently given to all other methods using Connector. Ideally all the other classes should be tagged "private"